sway/ipc-server.c

```diff
@@ -72,15 +72,12 @@ static void handle_display_destroy(struct wl_listener *listener, void *data) {
 	close(ipc_socket);
 	unlink(ipc_sockaddr->sun_path);
 
-	while (ipc_client_list->length) {
-		struct ipc_client *client = ipc_client_list->items[0];
-		ipc_client_disconnect(client);
+	for (int i = 0; i < ipc_client_list->length; ++i) {
+		ipc_client_disconnect(ipc_client_list->items[i]);
```

Comment:
`ipc_client_disconnect` removes the item from the list, which changes the length while you're iterating it. I'm pretty sure you want the while loop with `items[0]` instead of `items[i]`.

Reply:
Oh, I see. I'll revert this change then.